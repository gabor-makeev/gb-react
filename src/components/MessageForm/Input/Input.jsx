import style from './Input.module.scss';
import { useRef } from 'react';

export const Input = ({ isFocused = false, placeholder, value, setValue }) => {
  const element = useRef();

  if (isFocused) {
    element.current.focus();
  }

  return (
    <input
      type="text"
      value={value}
      ref={element}
      className={style.input}
      placeholder={placeholder}
      onChange={(e) => {
        setValue(e.target.value);
      }}
    />
  );
};
