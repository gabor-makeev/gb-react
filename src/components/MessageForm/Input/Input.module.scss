.input {
  padding: 15px;
  border: none;
  border-radius: 15px;
  background-color: #cccccc;
}