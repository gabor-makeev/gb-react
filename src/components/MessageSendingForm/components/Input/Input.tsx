import style from './Input.module.scss';
import { FC, useEffect, useRef } from 'react';
import TextField from '@mui/material/TextField';

interface InputProps {
  placeholder?: string;
  value?: string;
  setValue: (inputValue: string) => void;
  shouldAutoFocus?: boolean;
}

export const Input: FC<InputProps> = ({
  placeholder,
  value,
  setValue,
  shouldAutoFocus = true,
}) => {
  const element = useRef<HTMLInputElement | undefined>();

  useEffect(() => {
    if (shouldAutoFocus) {
      return element.current?.focus();
    }
  });

  return (
    <TextField
      label={'Message field'}
      variant="outlined"
      type="text"
      value={value}
      placeholder={placeholder}
      inputRef={element}
      className={style.input}
      onChange={(e) => {
        setValue(e.target.value);
      }}
    />
  );
};
