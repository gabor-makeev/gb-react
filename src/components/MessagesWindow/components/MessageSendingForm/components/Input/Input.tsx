import style from './Input.module.scss';
import { FC, useEffect, useRef } from 'react';

interface InputProps {
  placeholder?: string;
  disabled?: boolean;
  value?: string;
  setValue: (inputValue: string) => void;
  shouldAutoFocus?: boolean;
}

export const Input: FC<InputProps> = ({
  placeholder,
  disabled = false,
  value,
  setValue,
  shouldAutoFocus = true,
}) => {
  const element = useRef<HTMLInputElement>(null);

  useEffect(() => {
    if (shouldAutoFocus) {
      return element.current?.focus();
    }
  });

  return (
    <input
      type="text"
      value={value}
      disabled={disabled}
      placeholder={placeholder}
      ref={element}
      className={style.input}
      onChange={(e) => {
        setValue(e.target.value);
      }}
      data-testid={'messageSendingFormInput'}
    />
  );
};
