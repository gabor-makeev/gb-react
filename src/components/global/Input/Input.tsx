import { FieldContainer } from 'components/global/Input/components/FieldContainer/FieldContainer';
import { Svg } from 'components/global/Input/components/Svg/Svg';
import { Field } from 'components/global/Input/components/Field/Field';
import { Container } from 'components/global/Input/components/Container/Container';
import { FC } from 'react';

export enum InputTypes {
  text = 'text',
  email = 'email',
  password = 'password',
}

interface InputProps {
  inputType?: InputTypes;
  labelText?: string;
  svg?: JSX.Element;
  placeholder?: string;
}

export const Input: FC<InputProps> = ({
  inputType = InputTypes.text,
  labelText,
  svg,
  placeholder,
}) => {
  return (
    <Container>
      {labelText && <label htmlFor={'field'}>{labelText}</label>}
      <FieldContainer>
        {svg && (
          <label htmlFor={'field'}>
            <Svg {...svg.props}>{svg.props.children}</Svg>
          </label>
        )}
        <Field id={'field'} type={inputType} placeholder={placeholder} />
      </FieldContainer>
    </Container>
  );
};
