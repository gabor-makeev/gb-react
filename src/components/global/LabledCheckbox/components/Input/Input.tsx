import styled from 'styled-components';

export const Input = styled.input.attrs({
  type: 'checkbox',
})`
  visibility: hidden;
  height: 0;
  position: absolute;
`;
