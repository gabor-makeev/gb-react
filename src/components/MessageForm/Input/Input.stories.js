import { Input } from './Input';

export default {
  title: 'Input',
  componenet: Input,
};

const Template = (args) => <Input {...args} />;

export const Primary = Template.bind({});

export const WithPlaceholder = Template.bind({});
WithPlaceholder.args = {
  placeholder: 'placeholder',
};
