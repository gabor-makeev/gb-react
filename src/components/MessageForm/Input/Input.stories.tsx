import { Input } from './Input';
import { ComponentMeta, ComponentStory } from '@storybook/react';

export default {
  title: 'Input',
  componenet: Input,
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => (
  <Input shouldAutoFocus={false} {...args} />
);

export const Primary = Template.bind({});

export const WithPlaceholder = Template.bind({});
WithPlaceholder.args = {
  placeholder: 'placeholder',
};
