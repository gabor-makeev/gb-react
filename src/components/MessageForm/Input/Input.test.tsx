import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Input } from './Input';
import userEvent from '@testing-library/user-event';

describe('Input', () => {
  it('should render', () => {
    const mockHandler = jest.fn();

    render(<Input setValue={mockHandler} />);
  });

  it('should render with value', () => {
    const mockHandler = jest.fn();

    render(<Input setValue={mockHandler} value={'test value'} />);

    expect(screen.getByDisplayValue('test value')).toBeInTheDocument();
  });

  it('should render with placeholder', () => {
    const mockHandler = jest.fn();

    render(<Input setValue={mockHandler} placeholder={'test placeholder'} />);

    expect(screen.getByPlaceholderText('test placeholder')).toBeInTheDocument();
  });

  it('should render with snapshot', () => {
    const mockHandler = jest.fn();

    const { asFragment } = render(
      <Input
        setValue={mockHandler}
        placeholder={'test placeholder'}
        value={'test value'}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call setValue() on change', async () => {
    const mockHanlder = jest.fn();
    render(<Input setValue={mockHanlder} placeholder={'test placeholder'} />);

    const input = screen.getByPlaceholderText('test placeholder');
    await userEvent.type(input, 'test');

    expect(mockHanlder).toHaveBeenCalledTimes(4);
  });
});
