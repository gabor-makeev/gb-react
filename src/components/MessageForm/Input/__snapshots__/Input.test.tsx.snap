// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r3:"
      id=":r3:-label"
    >
      Message field
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r3:"
        placeholder="test placeholder"
        type="text"
        value="test value"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1in441m"
        >
          <span>
            Message field
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
